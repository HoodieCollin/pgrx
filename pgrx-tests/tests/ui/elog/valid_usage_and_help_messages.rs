use pgrx::prelude::*;

fn main() {}

// none of these should fail to compile

#[pg_extern]
fn simple_syntax_works() {
    log!("...");
    log!("...",);
    log!("{}", "...");
}

#[pg_extern]
fn ereport_simple_syntax_works() {
    ereport!(LOG, PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION, "...");
    ereport!(LOG, PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION, "...", "detail", "hint");
}

#[pg_extern]
fn alternative_syntax_works() {
    log!(
        message = "...";
    );
    log!(
        message = "...";
        detail = "detail";
    );
    log!(
        message = "...";
        detail = "detail";
        hint = "hint";
    );
    log!(
        message = "{}", "...";
    );
    log!(
        message = "{}", "...";
        detail = "{}", "detail";
    );
    log!(
        message = "{}", "...";
        hint = "{}", "hint";
    );
    log!(
        message = "{}", "...";
        detail = "{}", "detail";
        hint = "{}", "hint";
    );
}

#[pg_extern]
fn ereport_alternative_syntax_shorthand_works() {
    ereport! {
        loglevel = LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
    }
    ereport! {
        loglevel = LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
        detail = "detail";
    }
    ereport! {
        loglevel = LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
        hint = "hint";
    }
    ereport! {
        loglevel = LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
        detail = "detail";
        hint = "hint";
    }
}

#[pg_extern]
fn ereport_alternative_syntax_longhand_works() {
    ereport! {
        loglevel = PgLogLevel::LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
    }
    ereport! {
        loglevel = PgLogLevel::LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
        detail = "detail";
    }
    ereport! {
        loglevel = PgLogLevel::LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
        hint = "hint";
    }
    ereport! {
        loglevel = PgLogLevel::LOG;
        errcode = PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION;
        message = "...";
        detail = "detail";
        hint = "hint";
    }
}

#[pg_extern]
fn help_messages() {
    log!();
    info!();
    notice!();
    warning!();
    error!();
    debug1!();
    debug2!();
    debug3!();
    debug4!();
    debug5!();
    ereport!();
}
