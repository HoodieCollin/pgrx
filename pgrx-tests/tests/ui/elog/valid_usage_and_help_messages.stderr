error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         log!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         log! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:110:5
    |
110 |     log!();
    |     ^^^^^^
    |
    = note: this error originates in the macro `log` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         info!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         info! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:111:5
    |
111 |     info!();
    |     ^^^^^^^
    |
    = note: this error originates in the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         notice!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         notice! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:112:5
    |
112 |     notice!();
    |     ^^^^^^^^^
    |
    = note: this error originates in the macro `notice` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         warning!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         warning! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:113:5
    |
113 |     warning!();
    |     ^^^^^^^^^^
    |
    = note: this error originates in the macro `warning` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         error!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         error! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:114:5
    |
114 |     error!();
    |     ^^^^^^^^
    |
    = note: this error originates in the macro `error` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         debug1!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         debug1! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:115:5
    |
115 |     debug1!();
    |     ^^^^^^^^^
    |
    = note: this error originates in the macro `debug1` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         debug2!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         debug2! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:116:5
    |
116 |     debug2!();
    |     ^^^^^^^^^
    |
    = note: this error originates in the macro `debug2` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         debug3!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         debug3! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:117:5
    |
117 |     debug3!();
    |     ^^^^^^^^^
    |
    = note: this error originates in the macro `debug3` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         debug4!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         debug4! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:118:5
    |
118 |     debug4!();
    |     ^^^^^^^^^
    |
    = note: this error originates in the macro `debug4` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       # Simple Usage

       ## Required arguments:
         - string literal

       ## Optional arguments:
         - formatting arguments

       ## Example
         debug5!("{}", 42);

       # Alternative Usage

       ## Required arguments:
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Example
         debug5! {
           message = "...";
           detail = "extra detail";
           hint = "plus a hint";
         }

   --> tests/ui/elog/valid_usage_and_help_messages.rs:119:5
    |
119 |     debug5!();
    |     ^^^^^^^^^
    |
    = note: this error originates in the macro `debug5` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid invocation

       ## Required arguments:
         - loglevel: [PgLogLevel]
         - errcode: [PgSqlErrorCode]
         - message: impl Into<String>

       ## Optional arguments:
         - detail: impl Into<String>
         - hint: impl Into<String>

       ## Simple Usage
         ereport!(
           PgLogLevel::ERROR,
           PgSqlErrorCode::ERRCODE_INTERNAL_ERROR,
           "invalid input",
           "extra detail",
           "plus a hint",
         );

       ## Alternative Usage
         ereport! {{
           loglevel = PgLogLevel::ERROR;
           errcode = PgSqlErrorCode::ERRCODE_INTERNAL_ERROR;
           message = "invalid input";
           detail = "extra detail";
           hint = "plus a hint";
         }}

   --> tests/ui/elog/valid_usage_and_help_messages.rs:120:5
    |
120 |     ereport!();
    |     ^^^^^^^^^^
    |
    = note: this error originates in the macro `ereport` (in Nightly builds, run with -Z macro-backtrace for more info)
